starfish/core/spots/DecodeSpots/util.py

```diff
@@ -122,6 +126,449 @@ def _build_intensity_table(
     return intensity_table
 
 
+def _build_spot_traces_per_round(
+        round_dataframes: Dict[int, pd.DataFrame],
```

Comment:
```suggestion
        round_dataframes: Mapping[int, pd.DataFrame],
```

if you're not going to modify, might as well make it immutable.